### PR TITLE
Editor Consolidation

### PR DESCRIPTION
**Pull request by askvortsov1**
Companion to https://github.com/flarum/core/pull/2826

- Drop mdarea for now. The newest version has a lot of bugs that make it unusable, and it doesn't combine well with our editor system. In future releases, I'd like to include it in core (or at least support list continues and indent/outdent)
- Move github markdown toolbar stuff to core so it can be used by other extensions
- Massively simplify toolbar and button system.

### CODE REVIEW

**Review by tankerkiller125 — APPROVED** [review]

**Review by davwheat — APPROVED** [review]
Looks good, just that one idea. Can we wait for review comments on https://github.com/flarum/core/pull/2826 before merging though, as a fix for Win + Enter submitting could be used here to reduce complexity.

File: js/src/forum/components/MarkdownToolbar.js
Comment (dsevillamartin) [inline js/src/forum/components/MarkdownToolbar.js:5]: Is this change from `id` to `class` necessary? Are we intentionally breaking styles that use the ID here?

Comment (davwheat) [reply js/src/forum/components/MarkdownToolbar.js:5]: I do agree with this change, just in case theres two of these toolbars on screen at the same time, despite how ridiculously unlikely that would be

Comment (askvortsov1) [reply js/src/forum/components/MarkdownToolbar.js:5]: Iirc, there isn't ID based stuff anywhere else in the composer hierarchy. I don't think it should be here either.

**Review by davwheat — APPROVED** [review]
Reconfirming the approve because GitHub is GitHub

**Pull request merged**
